### PR TITLE
moar warning fixes

### PR DESCRIPTION
**Pull request by Morriar**

### CODE REVIEW

**Comment by privat** [comment]
+1

**Comment by privat** [comment]
test this please

**Comment by Morriar** [comment]
test this

**Comment by Morriar** [comment]
test this please

**Comment by Morriar** [comment]
test this please

**Comment by privat** [comment]
I need ack from maintainers like @xymus

File: contrib/benitlux/src/server/benitlux_db.nit [hunk 1]
```diff
@@ -173,7 +173,7 @@ DISTINCT day FROM daily WHERE beer=(SELECT ROWID FROM beers WHERE name="{{{beer.
 	fun subscribers: Array[String]
 	do
 		var subs = new Array[String]
-		for row in select("email FROM subscribers") do subs.add row[0].to_s
+		for row in select("email FROM subscribers").as(not null) do subs.add row[0].to_s
```
Comment (xymus) [inline contrib/benitlux/src/server/benitlux_db.nit]: This method should not assert. You are be better off to leave the warning as a reminder so I can fix it the next time I'll open this module.

File: contrib/objcwrapper/src/objc_visitor.nit [hunk 2]
```diff
@@ -77,7 +77,7 @@ redef class Nsignature_block_signature
 		assert c != null
 
 		var method = new ObjcMethod(c)
-		method.return_type = n_signature_return_type.to_type
+		method.return_type = n_signature_return_type.as(not null).to_type
```
Comment (xymus) [inline contrib/objcwrapper/src/objc_visitor.nit]: This looks like a bug. Either the grammar is wrong or it should be handled better. Unless you want to learn the Objective-C grammar, you can leave the warning.

File: lib/mpi/examples/src/mpi_simple.nit [hunk 3]
```diff
@@ -27,8 +27,7 @@ class ProcessorInfo
 	var name: String
 	var string_of_random_length: String
 
-	init(mpi: MPI)
-	do
+	init(mpi: MPI) is old_style_init do
```
Comment (xymus) [inline lib/mpi/examples/src/mpi_simple.nit]: Instead of adding the annotation, the constructor should be updated. And come on, don't change the style.

**Comment by xymus** [comment]
I don't like the idea of suppressing warnings just to hide them.

I'd rather be bugged by a `call on nullable` warning in my code until I fix it than than having a `as(not null)` crash at runtime. Even in case of a crash, the warning makes it easier to debug. Same with the warnings `attributes in refinement` and `old style init`, they indicate that it is old code and should be updated, so adding an annotation to suppress the warning hides potential problems and doesn't push us to update the examples.

File: contrib/benitlux/src/server/benitlux_db.nit (hunk 1 above)
Comment (Morriar) [inline contrib/benitlux/src/server/benitlux_db.nit]: Looking at the git blame for `contrib/benitlux/src/benitlux_db.nit`:

* `Alexis Laferrière 2014-07-18 02:31:36` code with warnings introduced
* `Alexis Laferrière 2015-08-06 15:30:58` code opened
* `Alexis Laferrière 2016-04-08 11:08:41` code opened again

I'm glad the warnings act as reminders. Do you even `nitc -W`?

**Comment by Morriar** [comment]
Seems like we are keeping those warnings then.

**Pull request closed without merging**